src/libNetwork/P2PComm.cpp

```diff
@@ -204,19 +210,74 @@ bool SendJob::SendMessageSocketCore(const Peer& peer, const bytes& message,
     serv_addr.sin_addr.s_addr = peer.m_ipAddress.convert_to<unsigned long>();
     serv_addr.sin_port = htons(peer.m_listenPortHost);
 
-    if (connect(cli_sock, (struct sockaddr*)&serv_addr, sizeof(serv_addr)) <
-        0) {
-      LOG_GENERAL(WARNING, "Socket connect failed. Code = "
-                               << errno << " Desc: " << std::strerror(errno)
-                               << ". IP address: " << peer);
-      if (P2PComm::IsHostHavingNetworkIssue()) {
-        LOG_GENERAL(WARNING, "[blacklist] Encountered "
-                                 << errno << " (" << std::strerror(errno)
-                                 << "). Adding " << peer.GetPrintableIPAddress()
-                                 << " to blacklist");
-        Blacklist::GetInstance().Add(peer.m_ipAddress);
+    // Set non-blocking
+    if ((arg = fcntl(cli_sock, F_GETFL, NULL)) < 0) {
+      LOG_GENERAL(WARNING, "could not get flags on TCP listening socket...");
+      return false;
+    }
+    arg |= O_NONBLOCK;
+    if (fcntl(cli_sock, F_SETFL, arg) < 0) {
+      LOG_GENERAL(WARNING,
+                  "could not set TCP listening socket to be non-blocking...");
+      return false;
+    }
+
+    int status;
+    bool connectStat = true;
+    if ((status = connect(cli_sock, (struct sockaddr*)&serv_addr,
+                          sizeof(serv_addr))) < 0) {
+      if (errno != EINPROGRESS) {
+        LOG_GENERAL(WARNING, "Error connecting(1), Cancelling connection!");
```

Comment:
Since we anticipate getting a lot of these messages, I suggest to make them very short.  Anyway, our log includes the line number, so it should be easy to know which failure was encountered.

Reply:
ok